### ISSUE TITLE
Duplicated intellisense messages following Julia language server crash

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/vscode-languageserver-node#246]
_From @ZacLN on August 2, 2017 13:19_

- VSCode Version: 1.4.1
- OS Version: Linux

When the julia-vscode language server plugin crashes and restarts intellisense messages from the server are duplicated.

Steps to Reproduce:

1. Install the julia-vscode extension: https://github.com/JuliaEditorSupport/julia-vscode/releases/tag/v0.7.0-beta.2 (requires julia v0.6)
2. Open a file,  switch the language to Julia and enter `rand()`, hover over `rand`
![julia-vscode1](https://user-images.githubusercontent.com/5861046/28875256-762394a6-7784-11e7-9756-47182edee7ae.png)
3. Simulate a crash by killing the language server (`ps -A |grep julia`, then `kill` _`pid`_)
4. wait for language server to restart and hover over `rand`
![julia-vscode2](https://user-images.githubusercontent.com/5861046/28875296-9f61bf50-7784-11e7-9872-9d58e64bbf82.png)

There is no difference in the message sent from language server but the client duplicates the display results by n (the number of times the server has crashed). 

We get the same duplication for other displayed language server responses.

_Copied from original issue: Microsoft/vscode#31915_

**Comment by dbaeumer** [comment]
_From @kieferrm on August 2, 2017 17:38_

@ZacLN how is the language client written? Do you use `vscode-languageclient` or is it custom?

**Comment by dbaeumer** [comment]
_From @ZacLN on August 2, 2017 17:47_

we use `"vscode-languageclient": "^3.0.0",`

[implementation](https://github.com/JuliaEditorSupport/julia-vscode)

**Comment by dbaeumer** [comment]
@ZacLN I tried to reproduce this in my test bed and I can't. Could I ask you to help me understand what is going on by debugging it locally. When a server crashes the `cleanup` method is called on the client (in file client.js). Can you set a breakpoint in there and step through

```
        for (let handler of this._registeredHandlers.values()) {
            handler.dispose();
        }
```

Do you hit this code?

**Comment by ZacLN** [comment]
In trying this I found that my local install had vs-languageclient@3.2.0, I can't replicate this having updated to 3.3.0

**Comment by dbaeumer** [comment]
Super. I fixed disposing the handlers in 3.3.x. Closing the issue then.

**Comment by ZacLN** [comment]
Ta
